rest/tests/test_rest_block.py

```diff
@@ -39,6 +39,8 @@ def __init__(self, events):
 
     def poll(self, paging=False):
         self._poll_events[0].set()
+        if len(self._poll_events) == 1:
+            return
```

Comment:
We agreed that this is ok for this test. A second test should be added for checking the retry_limit.